src/adapter/src/client.rs

```diff
@@ -483,6 +484,8 @@ impl SessionClient {
     {
         let session = self.session.take().expect("session invariant violated");
         let mut typ = None;
+        let application_name = session.vars().application_name();
```

Comment:
This should maybe just be a property and method on the session since it gets used enough. It'd have to know when that variable changes though to update itself so...maybe it's fine as is?

Reply:
Makes sense to me, I made a new method on `Session` for `application_name` like we have for `user`